src/pb_stub.cc

```diff
@@ -838,11 +820,45 @@ Stub::RunCoroutine(py::object coroutine, bool in_background)
     py_future.attr("add_done_callback")(
         py::module_::import("c_python_backend_utils")
             .attr("async_event_future_done_callback"));
+    background_futures_.attr("add")(py_future);
     return py::none();
   }
   return py_future.attr("result")();
 }
 
+void
+Stub::BackgroundFutureDone(const py::object& py_future)
```

Comment:
nit: I don't think we require a new function. Let's have the body in the `AsyncEventFutureDoneCallback`.